### ISSUE TITLE
Server HANGS when a new player joins

### ISSUE THREAD

**Issue by Drc-DEV** [BenCodez/VotingPlugin#513]
When an unregistered player joins this error comes up and the server hangs for 10 seconds

PaperSpigot 1.12.2 using VotingPlugin v5.13.6 (please make the latest version compatible with 1.12.2 with the new features.)
https://pastebin.com/4U1n1YrZ

**Comment by BenCodez** [comment]
What version are you using?

Does switching away from sqlite help?

**Comment by Drc-DEV** [comment]
I already wrote the version in my first post dude... Anyways I'll try with mysql right now. The best solution would be to make the latest version compatible with 1.12.2.

**Comment by BenCodez** [comment]
I meant plugin version (which I see you edited). 

Also, no. The best solution is to update to 1.13.2.

**Comment by Drc-DEV** [comment]
Mysql seems not to have that problem. Thanks for the suggestion.

**Comment by Drc-DEV** [comment]
The problem is still present, even when using MySQL database instead of SQLite

**Comment by BenCodez** [comment]
Are you using the latest version?

**Comment by Drc-DEV** [comment]
No, I am using the latest available version for 1.12.2, because my server and a lot of other servers can't update to 1.13.2 which has huge lag with mob-farms and chunk-gen.

**Comment by BenCodez** [comment]
Nothing I can do then.

**Comment by BenCodez** [comment]
Chances are it's fixed the newer versions anyway.

**Comment by Drc-DEV** [comment]
Can't you make a 1.12.2 working version of the latest code? Just one version to make 1.12.2 server owners still use VotingPlugin.

**Comment by BenCodez** [comment]
If you want to go through the effort then go ahead.

**Comment by Drc-DEV** [comment]
At least warn people downloading that 1.12.2 version you recommend on spigot that their server will get lag spikes when new players join. Thanks for being really nice and helpful.

**Comment by BenCodez** [comment]
Your the only one reporting this, plus most people are on 1.13 that currently use the plugin.
Plus it's an unsupported version.

**Comment by Drc-DEV** [comment]
https://github.com/Ben12345rocks/VotingPlugin/issues/431
Yeah the only one

Anyways, most of the problems with compatibility rely on your "AdvancedCore" which uses NMS in a couple of classes. And a couple of Materials here an there in both VotingPlugin and AdvancedCore, and the Advanced Tab in votingplugin, which could be easily toggled. It does not seem that much of a deal to make this plugin 1.12.2 compatible.

**Comment by Drc-DEV** [comment]
I managed to make a functioning 1.12 version from latest source code. The lag is still there when a new player joins. 

While debugging I found out the cause.. When I have "votingplugin" placeholderapi extension in use somewhere, it lags. Without the extension or without any placeholder being handled, it works fine, no lag.

So, as I'm using now latest code (with just auto-tab completion removed for 1.12 compatibility), the problem still persists, but it's related to placeholders someway.
